language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java

```diff
@@ -117,6 +128,100 @@ public CompletableFuture<ExpressionTypeResponse> type(ExpressionTypeRequest requ
         });
     }
 
+    @JsonRequest
+    public CompletableFuture<SymbolInfoResponse> getSymbol(SymbolInfoRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            SymbolInfoResponse symbolInfoResponse = new SymbolInfoResponse();
+            String fileUri = request.getDocumentIdentifier().getUri();
+            Optional<Path> filePath = CommonUtil.getPathFromURI(fileUri);
+
+            if (filePath.isEmpty()) {
+                return symbolInfoResponse;
+            }
+
+            try {
+                Optional<SemanticModel> semanticModel = this.workspaceManagerProxy.get(fileUri).
+                        semanticModel(filePath.get());
+                if (semanticModel.isEmpty()) {
+                    return symbolInfoResponse;
+                }
+
+                DocumentServiceContext context = ContextBuilder.buildDocumentServiceContext(fileUri,
+                        this.workspaceManagerProxy.get(fileUri),
+                        LSContextOperation.SYMBOL_DOCUMENT,
+                        this.serverContext);
+
+                Optional<Document> srcFile = context.currentDocument();
+                LinePosition linePosition = LinePosition.from(request.getPosition().getLine(),
+                        request.getPosition().getCharacter());
+                Optional<? extends Symbol> symbolAtCursor = semanticModel.get().symbol(srcFile.get(), linePosition);
+
+                if (symbolAtCursor.isEmpty()) {
+                    return symbolInfoResponse;
+                }
+
+                Optional<Documentation> documentation = symbolAtCursor.get() instanceof Documentable ?
+                        ((Documentable) symbolAtCursor.get()).documentation() : Optional.empty();
+
+                if (documentation.isEmpty()) {
+                    return symbolInfoResponse;
+                }
+
+                List<SymbolDocumentation.ParameterInfo> symbolParams = new ArrayList<>(documentation.get().
+                        parameterMap().size());
+
+                if (symbolAtCursor.get() instanceof FunctionSymbol) {
+                    FunctionSymbol functionSymbol = (FunctionSymbol) symbolAtCursor.get();
+
+                    if (functionSymbol.typeDescriptor().params().isPresent()) {
+                        List<ParameterSymbol> parameterSymbolList = functionSymbol.typeDescriptor().params().get();
+                        parameterSymbolList.forEach(parameterSymbol -> {
+                            symbolParams.add(new SymbolDocumentation.ParameterInfo(
+                                    parameterSymbol.getName().get(),
+                                    documentation.get().parameterMap().get(parameterSymbol.getName().get()),
```

Comment:
documentation.get().parameterMap().get(parameterSymbol.getName().get()) can return `null` in some cases. Shall we verify this using a filter